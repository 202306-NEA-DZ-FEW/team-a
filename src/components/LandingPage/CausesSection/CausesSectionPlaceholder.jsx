function CausesSectionPlaceholder() {
  return (
    <div className='flex flex-col flex-wrap gap-8 justify-center items-center bg-base-100 w-full min-h-screen p-8'>
      <div className='w-[300px] h-[30px] rounded-full bg-gray-300 animate-pulse mb-8'></div>
      <div className='w-full flex flex-wrap justify-center items-center gap-6 text-center'>
        {[1, 2, 3].map((i) => (
          <div
            key={i}
            className='flex-1 min-w-[300px] h-96 flex-grow bg-gray-300 animate-pulse rounded-2xl'
          ></div>
        ))}
      </div>
    </div>
  );
}

export default CausesSectionPlaceholder;
