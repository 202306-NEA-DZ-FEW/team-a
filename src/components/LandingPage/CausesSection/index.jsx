import { motion } from "framer-motion";
import { useTranslation } from "next-i18next";
import { FaHandHoldingWater } from "react-icons/fa";
import { LuHeartHandshake } from "react-icons/lu";
import { RiGraduationCapFill } from "react-icons/ri";

import Container from "@/components/container";

import InfoCard from "../InfoCard";

function CausesSection() {
  const { t } = useTranslation();
  const container = {
    hidden: { opacity: 1, scale: 0 },
    visible: {
      opacity: 1,
      scale: 1,
      transition: {
        delayChildren: 0.3,
        staggerChildren: 0.2,
      },
    },
  };

  const item = {
    hidden: { y: 20, opacity: 0 },
    visible: {
      y: 0,
      opacity: 1,
    },
  };
  return (
    <Container className='flex flex-col flex-wrap gap-8 justify-center lg:min-h-screen lg:my-0'>
      <motion.h1
        initial={{ y: 30, opacity: 0 }}
        whileInView={{ y: 0, opacity: 1 }}
        transition={{ duration: 0.7, ease: "easeInOut" }}
        className='text-3xl md:text-5xl gap-4 font-bold text-center'
      >
        {t("landingPage:causes:title")}
      </motion.h1>
      <motion.div
        variants={container}
        initial='hidden'
        whileInView='visible'
        className='w-full flex flex-wrap justify-center items-center gap-6 text-center'
      >
        <motion.div
          variants={item}
          transition={{ duration: 0.7, ease: "easeInOut" }}
          className='flex-1 min-w-[31%] h-96 flex-grow flex justify-center intems-center bg-base-100 drop-shadow-lg rounded-xl '
        >
          <InfoCard
            className='w-full h-full'
            icon={<RiGraduationCapFill className='text-6xl text-error' />}
            title={t("landingPage:causes:education:title")}
            description={t("landingPage:causes:education:description")}
          />
        </motion.div>
        <motion.div
          variants={item}
          transition={{ duration: 0.7, ease: "easeInOut" }}
          className='flex-1 min-w-[31%] h-96 flex-grow flex justify-center intems-center bg-base-100 drop-shadow-lg rounded-xl '
        >
          <InfoCard
            className='w-full h-full'
            icon={<LuHeartHandshake className='text-6xl text-error' />}
            title={t("landingPage:causes:solidarity:title")}
            description={t("landingPage:causes:solidarity:description")}
          />
        </motion.div>
        <motion.div
          variants={item}
          transition={{ duration: 0.7, ease: "easeInOut" }}
          className='flex-1 min-w-[31%] h-96 flex-grow flex justify-center intems-center bg-base-100 drop-shadow-lg rounded-xl '
        >
          <InfoCard
            className='w-full h-full'
            icon={<FaHandHoldingWater className='text-6xl text-error' />}
            title={t("landingPage:causes:health:title")}
            description={t("landingPage:causes:health:description")}
          />
        </motion.div>
      </motion.div>
    </Container>
  );
}

export default CausesSection;
