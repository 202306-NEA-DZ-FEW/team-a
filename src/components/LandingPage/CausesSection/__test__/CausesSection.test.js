import renderer from "react-test-renderer";

import CausesSection from "..";

it("renders correctly", () => {
  const tree = renderer.create(<CausesSection />).toJSON();
  expect(tree).toMatchSnapshot();
});
