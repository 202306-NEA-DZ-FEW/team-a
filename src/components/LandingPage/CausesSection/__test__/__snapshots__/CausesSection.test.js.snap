// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex flex-col flex-wrap gap-8 justify-center items-center bg-base-100 w-full min-h-screen p-8"
>
  <h2
    className="text-3xl py-10 md:text-5xl gap-4 font-black text-center"
    style={
      Object {
        "opacity": 0,
        "transform": "translateY(30px) translateZ(0)",
      }
    }
  >
    causes.title
  </h2>
  <div
    className="w-full flex flex-wrap justify-center items-center gap-6 text-center"
  >
    <div
      className="flex-1 min-w-[400px] h-96 flex-grow bg-base-100 flex justify-center intems-center bg-opacity-20 backdrop-filter backdrop-blur-md group rounded-3xl shadow-2xl overflow-hidden cursor-pointer"
      style={
        Object {
          "opacity": 0,
          "transform": "translateY(30px) scale(0) translateZ(0)",
        }
      }
    >
      <div
        className="relative  w-full h-full bg-cover bg-no-repeat flex flex-col flex-1 justify-center items-center gap-4"
        style={
          Object {
            "backgroundImage": "linear-gradient(rgba(114, 173, 132, 0.64),rgba(80, 140, 98, 1)), url(/img.jpg)",
          }
        }
      >
        <h2
          className="absolute bottom-8 left-24 origin-bottom-left -rotate-90 font-black text-white tracking-wider text-5xl"
        >
          causes.education.title
        </h2>
        <p
          className="absolute p-8 text-3xl font-light w-full h-full bg-neutral flex justify-center items-center translate-x-72 opacity-0 transition-all duration-700 ease-in-out group-hover:translate-x-0 group-hover:opacity-100 text-white/80"
        >
          causes.education.description
        </p>
      </div>
    </div>
    <div
      className="flex-1 min-w-[400px] h-96 flex-grow bg-base-100 flex justify-center intems-center bg-opacity-20 backdrop-filter backdrop-blur-md group rounded-3xl shadow-2xl overflow-hidden cursor-pointer"
      style={
        Object {
          "opacity": 0,
          "transform": "translateY(30px) scale(0) translateZ(0)",
        }
      }
    >
      <div
        className="relative  w-full h-full bg-cover bg-no-repeat flex flex-col flex-1 justify-center items-center gap-4"
        style={
          Object {
            "backgroundImage": "linear-gradient(rgba(114, 173, 132, 0.64),rgba(80, 140, 98, 1)), url(/img.jpg)",
          }
        }
      >
        <h2
          className="absolute bottom-8 left-24 origin-bottom-left -rotate-90 font-black text-white tracking-wider text-5xl"
        >
          causes.solidarity.title
        </h2>
        <p
          className="absolute p-8 text-3xl font-light w-full h-full bg-neutral flex justify-center items-center translate-x-72 opacity-0 transition-all duration-700 ease-in-out group-hover:translate-x-0 group-hover:opacity-100 text-white/80"
        >
          causes.solidarity.description
        </p>
      </div>
    </div>
    <div
      className="flex-1 min-w-[400px] h-96 flex-grow bg-base-100 flex justify-center intems-center bg-opacity-20 backdrop-filter backdrop-blur-md group rounded-3xl shadow-2xl overflow-hidden cursor-pointer"
      style={
        Object {
          "opacity": 0,
          "transform": "translateY(30px) scale(0) translateZ(0)",
        }
      }
    >
      <div
        className="relative  w-full h-full bg-cover bg-no-repeat flex flex-col flex-1 justify-center items-center gap-4"
        style={
          Object {
            "backgroundImage": "linear-gradient(rgba(114, 173, 132, 0.64),rgba(80, 140, 98, 1)), url(/img.jpg)",
          }
        }
      >
        <h2
          className="absolute bottom-8 left-24 origin-bottom-left -rotate-90 font-black text-white tracking-wider text-5xl"
        >
          causes.health.title
        </h2>
        <p
          className="absolute p-8 text-3xl font-light w-full h-full bg-neutral flex justify-center items-center translate-x-72 opacity-0 transition-all duration-700 ease-in-out group-hover:translate-x-0 group-hover:opacity-100 text-white/80"
        >
          causes.health.description
        </p>
      </div>
    </div>
  </div>
</div>
`;
